app/models/proxy_config.rb

```diff
@@ -121,6 +121,10 @@ def parsed_content
     JSON.parse(content).deep_symbolize_keys
   end
 
+  def contains_metric?(metric_id)
+    parsed_content[:proxy][:proxy_rules].pluck(:metric_id).include? metric_id
```

Comment:
Just a random thought. If we search by a regular expression would that be faster than parsing JSON, etc. especially when the config is big, like 1-2MB? But it's against humanity to parse structured data with regex. So purely a random thought.

Reply:
Hehe, yeah, I am not sure we'd gain much from it, as deleting a metric is not a very critical path. Unless it's a service deletion, that requires deleting all of the metrics, but in this case it looks like this validation will be skipped anyway because of https://github.com/3scale/porta/pull/3468/files#diff-7f8fd886a0f3b308a1c98cdef0ccb8ab908a2ad46244fb3b4dc7c0ca5152dab1R226 (not sure though). In any case - nothing terrible would happen I think.